### ISSUE TITLE
Incorrect color data coming out of rexpaintjs

### ISSUE THREAD

**Issue by NumeronR** [slashman/roguenet#4]
When I boot up a stock server, I get this:

![image](https://user-images.githubusercontent.com/19677926/125183747-2757d880-e258-11eb-9bdf-337f1c55fdb6.png)

The missing tiles are there, they're just black on black. I traced this back to server/LevelLoader.js where rexpaint's tile.fg objects are coming in with negatives instead of the expected 0-255 range.

Looking at rexpaint's buffer.fixedReadUint8 function in rexpainjs/index.js, there are a couple of commented out transformations. According to the comment, it seems like this function is supposed to clear the sign but doesn't - it returns signed ints when it should only return uints:
```
buffer.fixedReadUint8 = function(offset) {
      // There's a bug in readUint8 that damages the buffer, so
      // we read the two's complement and then clear the sign
      // with this terrible, terrible hack
      let v = this.readInt8(offset)
  
      return v //(v<0)?256+v:v //(v >>> 0) && 0xFF
    }
```
If I enable the first commented transformation `(v<0)?256+v:v`, the colors are now correct.

This it's such a severe bug I'm presuming it must work on your machine Slashie? The bug seems to lie in rexpaintjs, so did you make a change to that lib to get it working for you? Should we perhaps move the fix into LevelLoaded until (if ever) rexpaintjs is fixed? Or at least add a note to readme.md?

**Comment by slashman** [comment]
I checked this. It's related to https://github.com/chiguireitor/rexpaintjs/issues/1 . I am using the `(v<0)?256+v:v` transformation, manually hacked into my node_modules and probably the deployed server as well. The dev of the library will take an eye on this but for now we do need to use the hack so I suggest you activate it locally (we might decide to fork the lib with the hack itself and point to it in the package.json instead of the original but hopefully won't have to do that if it's fixed in the main lib)

**Comment by adri326** [comment]
So I looked into this a bit: the current version or chiguireitor/rexpaintjs has the transformation uncommented on version `0.1.2`, but your package-lock.json is still on `0.1.1`, so you should run `npm i rexpaintjs` to bump it and get the fix.

Moreover, the `fixedReadUInt8` function from rexpaintjs' source code addresses an issue that I couldn't reproduce (using `readUInt8` works just fine), maybe caused by an outdated node.js version. It worked on my end with `node` version `10.19.0` and version `16.3.0` without the fix. So I don't know why the function was there in the first place and why it's still there.

**Comment by slashman** [comment]
Fixed in b93a2372f2931bf093f0ecf5a6584c3c65a05b27
